x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -263,14 +263,13 @@ export const getXyVisualization = ({
   getDescription,
 
   switchVisualizationType(seriesType: string, state: State, layerId?: string) {
-    const dataLayer = state.layers.find((l) => l.layerId === layerId);
+    const dataLayer = state.layers.find((l) => l.layerId === layerId) ?? state.layers[0];
```

Comment:
nitpicking here: just to avoid picking a layer with an undefined `layerId` if ever exist.

```suggestion
    const dataLayer = layerId ? state.layers.find((l) => l.layerId === layerId) : state.layers.at(0);
```